launchable/test_runners/cucumber.py

```diff
@@ -140,31 +141,27 @@ def parse_func(self, report_file: str) -> Generator[CaseEventType, None, None]:
             click.echo("Can't find test reports from {}. Make sure to confirm report file.".format(
                 report_file), err=True)
 
+        background_test_case_info = None
         for d in data:
             file_name = d.get("uri", "")
             class_name = d.get("name", "")
             for element in d.get("elements", []):
                 test_case = element.get("name", "")
-                steps = {}
-                duration = 0
-                statuses = []
-                stderr = []
-                for step in element.get("steps", []):
-                    steps[step.get("keyword", "").strip()] = step.get("name", "").strip()
-                    result = step.get("result", None)
-
-                    if result:
-                        # duration's unit is nano sec
-                        # ref:
-                        # https://github.com/cucumber/cucumber-ruby/blob/main/lib/cucumber/formatter/json.rb#L222
-                        duration = duration + result.get("duration", 0)
-                        statuses.append(result.get("status"))
-                        if result.get("error_message", None):
-                            stderr.append(result["error_message"])
-
-                if "failed" in statuses:
+                element_type = element.get("type", "")
+                if element_type == CucumberElementType.BACKGROUND.value:
+                    background_test_case_info = _extract_test_case_info_from_elements(element=element)
+                    continue
+
+                test_case_info = _extract_test_case_info_from_elements(element=element)
+                if background_test_case_info:
+                    test_case_info.statuses += background_test_case_info.statuses
```

Comment:
I'm not sure if we should concatenate steps in background_test_case_info. Please let me know if you have an opinion.